wallet/how-to/use-non-evm-networks/starknet-snap/index.md

```diff
@@ -0,0 +1,174 @@
+This instance manages the Starknet account within MetaMask. With this setup, `get-starknet` enables the app to use Starknet features through MetaMask with the following capabilities:
+
+- `WalletAccount` uses a specified provider to read data from the Starknet network.
+- For transactions, `get-starknet` prepares the transaction data and sends it to MetaMask for signing through the Starknet Snap.
+- `get-starknet` allows the dapp to create contract instances connected to the `WalletAccount`. These instances can invoke smart contract functions, with MetaMask handling the necessary signatures.
+- `get-starknet` sets up listeners for account and network changes within MetaMask.
+The dapp can subscribe to these events to update its state accordingly.
+- `get-starknet` can request network changes through MetaMask, and allows users to switch between different Starknet networks (for example Mainnet, testnet).
+- `get-starknet` can request MetaMask to display specific tokens in its interface, enhancing user experience.
+
+
+```mermaid
+sequenceDiagram
+    participant dapp as dapp
+    participant get as get-starknet
+    participant mm as MetaMask
+    participant Snap as Starknet Snap
+    participant network as Starknet Network
+
+    dapp->>get: Initialize connection
+    get->>mm: Request connection
+    mm->>Snap: Activate (if needed)
+    Snap-->>mm: Activated
+    mm-->>get: Return SWO
+    get->>get: Create WalletAccount
+    get-->>dapp: Connection established
+    dapp->>get: Read blockchain data
+    get->>network: Query data
+    network-->>get: Return data
+    get-->>dapp: Processed data
+    dapp->>get: Write transaction
+    get->>mm: Request signature
+    mm->>Snap: Sign transaction
+    Snap-->>mm: Signed transaction
+    mm-->>get: Return signature
+    get->>network: Submit transaction
+    network-->>get: Transaction result
+    get-->>dapp: Transaction confirmation
+    mm->>get: Account/Network changed
+    get->>dapp: Notify change
+```
+
+
+## Compare wallet integration methods for Starknet
+
+You can use `wallet_invokeSnap` or `get-starknet` to integrate with Starknet dapps.
+The choice depends on your specific use case and development preferences. 
+
+| Feature | `wallet_invokeSnap` | `get-starknet` |
+|---------|---------------------|----------------|
+| API Level | Low-level access to Starknet Snap methods | High-level abstractions for Starknet operations |
+| Wallet Support | MetaMask only | Multiple Starknet-compatible wallets |
+| Functionality | Limited to Snap-implemented methods | Comprehensive toolkit for Starknet dapp development |
+| Multi-wallet Sync | No | Yes |
+| Standardization | Might lead to non-standard implementations | Enforces standardized wallet interactions |
+| Flexibility | High, with direct access to all Snap features | Moderate, within implemented features |
+| Code Portability | Low, tied to MetaMask | High, works across multiple wallet implementations |
```

Comment:
I think move this table higher up on this page so users can immediately compare the actual features of each method. Shorten up the intro and add this there instead.